sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/jdbc/nativejdbc/provider/AccessTokenProvider.java

```diff
@@ -0,0 +1,6 @@
+package com.azure.spring.cloud.autoconfigure.jdbc.nativejdbc.provider;
+
+
+interface AccessTokenProvider<T> {
```

Comment:
I think there are couple issues with this class

1. why this has a generic type? is there any other type provided by the `getAccessToken` than `AccessToken`?
2. even if it is, this class would be  better to extent from `Supplier` and makr as `FunctionalInterface`
3. `provider` from package seems meaningless